src/main/java/org/apache/sysml/hops/AggUnaryOp.java

```diff
@@ -329,8 +329,15 @@ public boolean allowsAllExecTypes()
 
 	@Override
 	protected double computeOutputMemEstimate( long dim1, long dim2, long nnz )
-	{		
-		double sparsity = OptimizerUtils.getSparsity(dim1, dim2, nnz);
+	{
+		double sparsity = -1;
+		if (DMLScript.USE_ACCELERATOR) {
+			// The GPU version (for the time being) only does dense outputs
+			sparsity = 1.0;
+		} else {
+			sparsity = OptimizerUtils.getSparsity(dim1, dim2, nnz);
+		}
```

Comment:
First, please use a single-line statements for such code (double sparsity=DMLScript.USE_ACCELERATOR ? 1 : OptimizerUtils.getSparsity(dim1, dim2, nnz)) - it's takes unnecessarily long to read this.

Second, and as a general note - it's fine to use internally dense outputs but these outputs need to be converted to sparse for any subsequent operations. Hence, the output memory estimate cannot be modified here because it is the input memory estimate of data-dependent operations (potentially executed in CP).